base/io.jl

```diff
@@ -113,7 +113,10 @@ flush(io::AbstractPipe) = flush(pipe_writer(io))
 read(io::AbstractPipe, byte::Type{UInt8}) = read(pipe_reader(io), byte)
 unsafe_read(io::AbstractPipe, p::Ptr{UInt8}, nb::UInt) = unsafe_read(pipe_reader(io), p, nb)
 read(io::AbstractPipe) = read(pipe_reader(io))
-readuntil{T<:AbstractPipe}(io::T, args...) = readuntil(pipe_reader(io), args...)
+readuntil(io::AbstractPipe, arg::UInt8)          = readuntil(pipe_reader(io), arg)
+readuntil(io::AbstractPipe, arg::Char)           = readuntil(pipe_reader(io), arg)
+readuntil(io::AbstractPipe, arg::AbstractString) = readuntil(pipe_reader(io), arg)
+readuntil(io::AbstractPipe, arg)                 = readuntil(pipe_reader(io), arg)
```

Comment:
is that last one going to always just be causing ambiguity issues? it's intended to help dispatch to a faster version of `readuntil{T}(io::AbstractPipe, arg::T)`.  but we can just drop it.